src/main/java/net/fabricmc/loader/discovery/ModResolver.java

```diff
@@ -253,7 +252,7 @@ private static IVecInt toVecInt(IntStream stream) {
 		StringBuilder errorsSoft = new StringBuilder();
 
 		if (!missingMods.isEmpty()) {
-			errorsHard.append("\n - Missing mods: ").append(Joiner.on(", ").join(missingMods));
+			errorsHard.append("\n - Missing mods: ").append(missingMods.stream().collect(Collectors.joining(", ")));
```

Comment:
Why not `String.join(", ", missingMods)`

Reply:
```suggestion
			errorsHard.append("\n - Missing mods: ").append(String.join(", ", missingMods));
```

Reply:
I replaced one with String.join (didn't know about that one yet..) and two that also had [] with Arrays.toString.